src/pass/split_host_device.cc

```diff
@@ -153,7 +153,9 @@ class HostDeviceSplitter : public IRMutator {
 
   Stmt Mutate_(const AttrStmt *op, const Stmt& s) final {
     if (op->attr_key == attr::thread_extent ||
-        op->attr_key == attr::pipeline_exec_scope) {
+        op->attr_key == attr::pipeline_exec_scope ||
+        (op->attr_key == attr::pragma_scope_prefix &&
+        op->value.as<StringImm>()->value == "device")) {
```

Comment:
in our current convention, we use attr=pragma_xyz to indicate a special pragma, so if we want to do this, we should use pragma_device in the key and ignore the value(can be just constant 0)